soda_machine.rb

```diff
@@ -7,12 +7,20 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.count
   end
 
   def find_soda(soda_brand)
+    @sodas.find { |soda| soda.brand == soda_brand }
   end
 
   def sell(soda_brand)
+    # @sodas.find { |soda| soda.brand == soda_brand }
+    if find_soda(soda_brand)
+      sold_soda = find_soda(soda_brand) 
```

Comment:
You're calling find_soda twice -- doing the same work again!  How could you avoid that?

Address this and push up a new commit.  When you do, tag me (@salogel42) so I can take another look!